### PR TITLE
Add support for historical balance lookups in Rosetta

### PR DESCRIPTION
**Pull request by omerzach**
Depends on https://github.com/MinaProtocol/mina/pull/7680

Adds support for historical balance lookups in Rosetta via queries to the balance tables in the archive DB.

Testing:
- Ran `./start.sh` and saw it passes
- Ran Rosetta CLI and saw that it still passes all past conditions and still fails on time-locked accounts

TODO:
- Write additional tests for `./start.sh` suite
- Waiting to hear back from the Rosetta team regarding this edge case https://community.rosetta-api.org/t/historical-balance-requests-with-block-identifiers-from-before-account-was-created/369 (but I think we can merge without it)

Checklist:

- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues?

### CODE REVIEW

File: src/app/rosetta/lib/dune [hunk 1]
```diff
@@ -26,6 +26,7 @@
  (instrumentation (backend bisect_ppx))
  (preprocess (pps
                graphql_ppx
+               h_list.ppx
```
Comment (omerzach) [inline src/app/rosetta/lib/dune:29]: I don't think I need this anymore?

**Comment by bkase** [comment]
!ci-build-me

**Review by bkase — APPROVED** [review]

File: src/app/rosetta/lib/account.ml [hunk 2]
```diff
@@ -29,13 +33,208 @@ module Get_balance =
     }
 |}]
 
+module Sql = struct
+  module Balance_from_last_relevant_command = struct
+    let query =
+      Caqti_request.find_opt
+        Caqti_type.(tup2 string int64)
+        Caqti_type.(tup2 int64 int64)
+        (* TODO: Lost (uc.type = 'delegation' OR uc.type = 'payment'), is that okay? *)
```
[the remark this answers is not in the document]
Comment (bkase) [reply src/app/rosetta/lib/account.ml:42]: Maybe other user commands will work still? @mrmr1993 will know

**Comment by bkase** [comment]
!ci-build-me

File: src/app/rosetta/lib/account.ml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mrmr1993) [reply src/app/rosetta/lib/account.ml:42]: Looks like it should just work to me!

**Comment by bkase** [comment]
!ci-build-me
